typescript/cli/src/config/hooks.ts

```diff
@@ -0,0 +1,228 @@
+    } else {
+      throw new Error('Cannot estimate gas overhead for IGP hook');
+    }
+    acc[chain] = multisigIsmVerificationCost(
+      validatorThreshold,
+      validatorCount,
+    );
+    return acc;
+  }, {});
+
+  return {
+    required: {
+      type: HookType.PROTOCOL_FEE,
+      maxProtocolFee: ethers.utils.parseUnits('1', 'gwei'),
+      protocolFee: ethers.utils.parseUnits('0', 'wei'),
+      beneficiary: owner,
+      owner: owner,
+    } as ProtocolFeeHookConfig,
+    default: {
+      type: HookType.AGGREGATION,
+      hooks: [
+        {
+          type: HookType.MERKLE_TREE,
+        } as MerkleTreeHookConfig,
+        {
+          type: HookType.INTERCHAIN_GAS_PAYMASTER,
+          owner: owner,
+          beneficiary: owner,
+          gasOracleType,
+          overhead,
+          oracleKey: owner,
+        } as IgpHookConfig,
+      ],
+    },
+  };
+}
+
+export function readHookConfig(filePath: string) {
+  const config = readYamlOrJson(filePath);
+  if (!config) {
+    logRed(`No multisig config found at ${filePath}`);
+    return;
+  }
+  const result = HookConfigMapSchema.safeParse(config);
+  if (!result.success) {
+    const firstIssue = result.error.issues[0];
+    throw new Error(
+      `Invalid hook config: ${firstIssue.path} => ${firstIssue.message}`,
+    );
+  }
+  const parsedConfig = result.data;
+  const defaultHook: ChainMap<HookConfig> = objMap(
+    parsedConfig,
+    (_, config) =>
+      ({
+        type: config.default.type,
+      } as HookConfig),
+  );
+  logGreen(`All multisig configs in ${filePath} are valid`);
+  return defaultHook;
+}
+
+// TODO: read different hook configs
+// export async function readProtocolFeeHookConfig(config: {type: HookType.PROTOCOL_FEE, ...}) {
+
+export async function createHookConfig({
+  format,
+  outPath,
+  chainConfigPath,
+}: {
+  format: FileFormat;
+  outPath: string;
+  chainConfigPath: string;
+}) {
+  logBlue('Creating a new hook config');
+  const customChains = readChainConfigIfExists(chainConfigPath);
+  const chains = await runMultiChainSelectionStep(customChains);
+
+  const result: HookConfigMap = {};
+  for (const chain of chains) {
+    for (const hookRequirements of ['required', 'default']) {
+      log(`Setting ${hookRequirements} hook for chain ${chain}`);
+      const hookType = await select({
+        message: 'Select hook type',
+        choices: [
+          { value: 'merkle_tree', name: 'MerkleTreeHook' },
+          { value: 'protocol_fee', name: 'StaticProtocolFee' },
+        ],
+        pageSize: 5,
+      });
+      switch (hookType) {
+        case 'merkle_tree': {
+          result[chain] = {
+            ...result[chain],
+            [hookRequirements]: { type: HookType.MERKLE_TREE },
+          };
+          break;
+        }
+        case 'protocol_fee': {
+          const owner = await input({
+            message: 'Enter owner address',
+          });
+          const ownerAddress = ethers.utils.getAddress(owner);
```

Comment:
Let's use our existing address related utils for this (`utils/src/addresses`). I'll fix now.